File: app/src/main/java/com/example/unlone/ui/Lounge/FragmentAdapter.java
package com.example.unlone.ui.Lounge;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class FragmentAdapter extends FragmentStateAdapter {
    public FragmentAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {

        switch (position){
            case 1:
                return new LoungeAllFragment();
            case 2:
                return new LoungeCategoryFragment();
        }

        return new LoungeFollowingFragment();
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/com/example/unlone/ui/lounge/FragmentAdapter.kt
package com.example.unlone.ui.lounge

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter

class FragmentAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
    FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment {
        when (position) {
            1 -> return LoungeAllFragment()
            2 -> return LoungeCategoryFragment()
        }
        return LoungeFollowingFragment()
    }

    override fun getItemCount(): Int {
        return 3
    }
}